IdentityCore/src/cache/MSIDTokenCacheKey.m

```diff
@@ -169,4 +156,134 @@ + (NSString *)familyClientId:(NSString *)familyId
     return [NSString stringWithFormat:@"foci-%@", familyId];
 }
 
+- (BOOL)isEqualToTokenCacheKey:(MSIDTokenCacheKey *)key
+{
+    if (!key)
+    {
+        return NO;
+    }
+    
+    BOOL result = YES;
+    result &= (!self.account && !key.account) || [self.account isEqualToString:key.account];
+    result &= (!self.service && !key.service) || [self.service isEqualToString:key.service];
+    result &= (!self.type && !key.type) || [self.type isEqualToNumber:key.type];
+    
+    return result;
+}
+
+#pragma mark - NSObject
+
+- (BOOL)isEqual:(id)object
+{
+    if (self == object)
+    {
+        return YES;
+    }
+    
+    if (![object isKindOfClass:MSIDTokenCacheKey.class])
+    {
+        return NO;
+    }
+    
+    return [self isEqualToTokenCacheKey:(MSIDTokenCacheKey *)object];
+}
+
+- (NSUInteger)hash
+{
+    NSUInteger hash = self.account.hash;
```

Comment:
Simply XORing values together is not a particularly good hashing function, the simple reason why is because if two values are the same they'll end up canceling each other out, resulting in considerably more hash collisions. This SO answer covers the topic more: https://stackoverflow.com/questions/1646807/quick-and-simple-hash-code-combinations/1646913#1646913

Reply:
changed to alg with prime numbers.